### ISSUE TITLE
KtLint throws an exception on trailing parameter default on newline

### ISSUE THREAD

**Issue by ZacSweers** [pinterest/ktlint#1437]
## Expected Behavior

It should not throw an exception

## Observed Behavior

it does!

## Steps to Reproduce

This snippet
```kotlin
@MustUseNamedParams
public fun copy(
  rankingModel: String? = this.rankingModel,
  topTermsCounts: Map<String, Int>? =
    this.topTermsCounts // <--- This is the problem line
): SearchRankingDebug = SearchRankingDebug(rankingModel, topTermsCounts)
```

```
Step 'ktlint' found problem in 'src/main/kotlin/slack/api/methods/search/modules/CommonSchemaDefs.kt':
Error on line: 300, column: 6
Trailing space(s)
java.lang.AssertionError: Error on line: 300, column: 6
Trailing space(s)
```

## Your Environment

* Version of ktlint used: 0.45.1 (but also reproducible in 0.44.0)
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Spotless 6.3.0
* Version of Gradle used (if applicable):
* Operating System and version: macos monterey

**Comment by ZacSweers** [comment]
False alarm! my test on 0.45.1 wasn't actually updating the version and it does appear to be fixed there
